### ISSUE TITLE
Support TargetRubyVersion: 1.8

### ISSUE THREAD

**Issue by segiddins** [bbatsov/rubocop#2640]
Some projects (Bundler among them) still have to support 1.8.7 for the foreseeable future. Parser supports 1.8, so it would be really nice if RuboCop did as well.

**Comment by bbatsov** [comment]
I'm sorry but this is not going to happen. We dropped support for 1.8 a while back and it's not coming back. I've been advising @whitequark to do the same for Parser in the interest of pushing people to accept the fact that 1.8 (and 1.9) are more or less dead.

**Comment by whitequark** [comment]
FWIW my position is that Parser will support 1.8 for as long as we can test it, i.e. as long as jruby-18mode exists, so it's unlikely to go anywhere ever.

**Comment by segiddins** [comment]
Well, thanks for saving me the time contributing this, I guess... That's kinda annoying that we can't use the `TargetRubyVersion` feature, though

**Comment by alexdowad** [comment]
> Some projects (Bundler among them) still have to support 1.8.7 for the foreseeable future. Parser supports 1.8, so it would be really nice if RuboCop did as well.

In my original `TargetRubyVersion` patch, 1.8 was supported, and the patch was explicitly not accepted for that reason.

It's quite easy to add support in your own little fork if you want to. Just about 6 lines of code, I think.

**Comment by bbatsov** [comment]
People need incentive to upgrade from time to time. :-) Btw, some of the cops probably do transformations that don't work on 1.8, so just adding this target won't be quite enough. Haven't used 1.8 for so long I don't even remember the differences. I just want people to keep this in mind.

**Comment by segiddins** [comment]
@bbatsov unfortunately, Bundler takes backwards compatibility and SemVer very very seriously, so we can't drop 1.8.7 until 2.0

**Comment by bbatsov** [comment]
Well, they should have issued it a long time ago IMO. I take very seriously Ruby's life cycle and have a strict policy not to support EOL Rubies. Support for 1.9 is still around mostly because so far, there hasn't been some compelling reason to drop it. Our initial plan was to kill it months ago.

**Comment by mockdeep** [comment]
One case for adding support is in helping to rehab older projects. I've adopted an [ancient Rails project](https://github.com/mockdeep/better) that's running 1.8.7 and it's going to be some time before we can bring it up to more recent versions of Ruby, so it would be nice to have support there to make the transition easier.

Side note: I don't think rubocop has ever supported 1.8.7. I've installed all the way back to `0.1.0` and it throws an error `undefined method `realpath' for File:Class`. It looks like `realpath` was added to `File` in `1.9`.

**Comment by bbatsov** [comment]
It never ran on 1.8, but it used to support parsing 1.8 code. I don't recall when we dropped the support for this.

**Comment by Drenmi** [comment]
> [...] it's going to be some time before we can bring it up to more recent versions of Ruby, so it would be nice to have support there to make the transition easier.

@mockdeep: What are the biggest obstacles to upgrading? Old Rails interoperability? Sincere question, since I'm curious. 🙂

**Comment by mockdeep** [comment]
@Drenmi The first biggest obstacle is not knowing whether things still work when I upgrade. When I forked this project, it had almost nothing in terms of test coverage, and pretty sure they weren't even passing at that point. There are a lot of, uh... interesting things going on in the codebase, so my first goal is to get the test coverage up. It's a pretty sizable codebase, though, so it's going to take a while.
